arangod/RestServer/Metrics.h

```diff
@@ -100,29 +100,43 @@ template<typename T> class Gauge : public Metric {
   ~Gauge() = default;
   std::ostream& print (std::ostream&) const;
   Gauge<T>& operator+=(T const& t) {
-    _g.store(_g + t);
+    T tmp;
+    do {
+      tmp = _g.load(std::memory_order_acquire);
+    } while (!_g.compare_exchange_weak(tmp, tmp + t, std::memory_order_acq_rel,
```

Comment:
After some discussion, @mpoeter and I agreed it would be better to handle these changes in a separate PR (#13047) with the memory orders parameterized and defaulted to sequential consistency.